tests/src/Kernel/EntityMemoryCacheTest.php

```diff
@@ -0,0 +1,245 @@
+      $this->assertNotContains($developer->getEmail(), $this->entityIdCache->getIds());
+    }
+
+    $this->assertEmptyCaches();
+  }
+
+  /**
+   * Tests app entity memory cache.
+   */
+  public function testAppEntityMemoryCache() {
+    $this->entityCache = $this->container->get('apigee_edge.controller.cache.apps');
+    $this->entityIdCache = $this->container->get('apigee_edge.controller.cache.app_ids');
+
+    // Generate developer app entities with random data.
+    $developer_apps = [];
+    $parent_id = $this->getRandomUniqueId();
+    for ($i = 0; $i < 2; $i++) {
+      $id = $this->getRandomUniqueId();
+      $developer_apps[$id] = new DeveloperApp([
+        'appId' => $id,
+        'name' => $this->getRandomUniqueId(),
+        'developerId' => $parent_id,
+      ]);
+    }
+
+    // Generate company app entities with random data.
+    $company_apps = [];
+    $parent_id = $this->getRandomUniqueId();
+    for ($i = 0; $i < 2; $i++) {
+      $id = $this->getRandomUniqueId();
+      $company_apps[$id] = new CompanyApp([
+        'appId' => $id,
+        'name' => $this->getRandomUniqueId(),
+        'companyName' => $parent_id,
+      ]);
+    }
+
+    $apps = $developer_apps + $company_apps;
+
+    $this->assertFullCaches($apps);
+
+    /** @var \Apigee\Edge\Api\Management\Entity\AppInterface $app */
+    foreach ($apps as $app) {
+      // Load app by id and by owner.
+      $this->assertSame($app, $this->entityCache->getEntity($app->getAppId()));
+      $this->assertContains($app, $this->entityCache->getAppsByOwner($this->entityCache->getAppOwner($app)));
+      $this->assertContains($app->getAppId(), $this->entityIdCache->getIds());
+    }
+
+    // Remove apps from cache by owner.
+    $this->entityCache->removeAppsByOwner(reset($developer_apps)->getDeveloperId());
+    $this->entityCache->removeAppsByOwner(reset($company_apps)->getCompanyName());
+
+    foreach ($apps as $app) {
+      $this->assertNull($this->entityCache->getEntity($app->getAppId()));
+      $this->assertNull($this->entityCache->getAppsByOwner($this->entityCache->getAppOwner($app)));
+      $this->assertNotContains($app->getAppId(), $this->entityIdCache->getIds());
+    }
+
+    $this->assertEmptyCaches();
+  }
+
+  /**
+   * Tests developer app entity memory cache.
+   */
+  public function testDeveloperAppEntityMemoryCache() {
+    $developer_app_cache_factory = $this->container->get('apigee_edge.entity.controller.cache.developer_app_cache_factory');
+
+    // Owner of the developer apps.
+    $developer = new Developer([
+      'developerId' => $this->getRandomUniqueId(),
+      'email' => strtolower($this->randomMachineName()) . '@example.com',
+    ]);
+    $developer_cache = $this->container->get('apigee_edge.controller.cache.developer');
+    $developer_cache->saveEntities([$developer]);
+
+    // Generate developer app entities with random data.
+    $developer_apps = [];
+    $id = $this->getRandomUniqueId();
```

Comment:
For easier maintenance and readability I'd rather fill in the developer_app array in a loop with these two entities and then use `list()` to explode this array to these two developer_app_[0-9] variables.